updater/internal/rollback/rollback.go

```diff
@@ -86,18 +91,20 @@ func (r Rollbacker) Backup() error {
 
 // Rollback performs a rollback by undoing all recorded actions.
 func (r Rollbacker) Rollback() {
+	r.logger.Debug("Performing rollback")
 	// We need to loop through the actions slice backwards, to roll back the actions in the correct order.
 	// e.g. if StartService was called last, we need to stop the service first, then rollback previous actions.
 	for i := len(r.actions) - 1; i >= 0; i-- {
 		action := r.actions[i]
+		r.logger.Debug("Rolling back action", zap.Any("action", action))
 		if err := action.Rollback(); err != nil {
-			log.Default().Printf("Failed to run rollback option: %s", err)
+			r.logger.Error("Failed to run rollback action", zap.Error(err))
 		}
 	}
 }
 
 // copyFiles copies files from inputPath to output path, skipping tmpDir.
-func copyFiles(inputPath, outputPath, tmpDir string) error {
+func (r Rollbacker) copyFiles(inputPath, outputPath, tmpDir string) error {
```

Comment:
Same here if we don't need this to be a receiver function can we just pass in the logger?